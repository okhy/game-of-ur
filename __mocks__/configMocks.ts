import { rowSetType } from "../src/types";

export const mockRowSet: rowSetType = [
  [
    {
      id: 1,
      pieces: [],
      accepts: []
    }
  ]
];
