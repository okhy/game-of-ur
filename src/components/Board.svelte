<script>
  import Tile from "./Tile.svelte";
  export let rowSet; // two dimensional array
</script>

<style>
  .row {
    display: flex;
  }
</style>

<div class="board">
  {#each rowSet as row}
    <div class="row">
      {#each row as tile}
        <Tile {...tile} />
      {/each}
    </div>
  {/each}
</div>
