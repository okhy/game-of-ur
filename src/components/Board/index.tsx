import React from 'react'

const Board: React.SFC = () => {

}

export default Board