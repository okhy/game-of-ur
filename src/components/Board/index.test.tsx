import React from "react";
import { shallow } from "enzyme";
// mocks
import { mockRowSet } from "./../../../__mocks__/configMocks";
// components
import Board from "./";

describe("Board component ...", () => {
  it("... renders without crashing", () => {
    const wrapper = shallow(<Board />);
    expect(wrapper).toBeTruthy();
  });

  it("... matches snapshot", () => {
    const wrapper = shallow(<Board />);
    expect(wrapper).toMatchSnapshot();
  });

  it("... renders tiles according to config", () => {
    const wrapper = shallow(<Board tileSet={mockRowSet} />);

    expect(wrapper.find(".row")).toHaveLength(1);
    expect(wrapper.find(".tile")).toHaveLength(1);
  });
});
