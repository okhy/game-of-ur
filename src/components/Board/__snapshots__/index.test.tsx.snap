// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board component ... ... matches snapshot 1`] = `ShallowWrapper {}`;
